_stbt/core.py

```diff
@@ -2527,21 +2526,33 @@ def _tesseract(frame, region, mode, lang, _config,
     else:
         region = intersection
 
-    return (_tesseract_subprocess(_crop(frame, region), mode, lang,
-                                  _config, user_patterns, user_words),
+    return (_tesseract_subprocess(_crop(frame, region), mode, lang, _config,
+                                  user_patterns, user_words, text_color),
             region)
 
 
 @imgproc_cache.memoize({"tesseract_version": str(_tesseract_version()),
-                        "version": "25"})
+                        "version": "27"})
 def _tesseract_subprocess(
-        frame, mode, lang, _config, user_patterns, user_words):
+        frame, mode, lang, _config, user_patterns, user_words, text_color):
+
     # We scale image up 3x before feeding it to tesseract as this
     # significantly reduces the error rate by more than 6x in tests.  This
     # uses bilinear interpolation which produces the best results.  See
     # http://stb-tester.com/blog/2014/04/14/improving-ocr-accuracy.html
     outsize = (frame.shape[1] * 3, frame.shape[0] * 3)
-    subframe = cv2.resize(frame, outsize, interpolation=cv2.INTER_LINEAR)
+    frame = cv2.resize(frame, outsize, interpolation=cv2.INTER_LINEAR)
+
+    if text_color is not None:
+        # Calculate distance of each pixel from `text_color`, then discard
+        # everything further than `threshold` distance away.
+        diff = frame - text_color
+        frame = numpy.sqrt((diff[:, :, 0] ** 2 +
+                            diff[:, :, 1] ** 2 +
+                            diff[:, :, 2] ** 2) / 3) \
+                     .astype(numpy.uint8)
+        threshold = 25
+        _, frame = cv2.threshold(frame, threshold, 255, cv2.THRESH_BINARY)
```

Comment:
This means that pixels close to the text color will go to black, and pixels far away from the text color will go to white. This is the right way round for tesseract (it likes black text on white backgrounds).